app/uk/gov/hmrc/cgtpropertydisposalsfrontend/models/finance/MoneyUtils.scala

```diff
@@ -54,6 +54,26 @@ object MoneyUtils {
         else Right(d)
       }
 
+  def validateLimit(key: String, exceedsLimit: BigDecimal => Boolean)(
+    s: String
+  ): Either[FormError, BigDecimal] =
+    Try(BigDecimal(cleanupAmountOfMoneyString(s))).toEither
+      .leftMap(_ => FormError(key, "error.invalid"))
+      .flatMap { d =>
+        if (exceedsLimit(d)) Left(FormError(key, "error.amountOverLimit"))
+        else Right(d)
+      }
+
+  def validateLimitLessThanOther(key: String, comparedOtherKey: String, exceedsLimit: BigDecimal => Boolean)(
```

Comment:
this is doing the same thing as `validateLimit` - you could have a general method like this:
```scala
 def validateAmountOfMoney(key: String, isValid: BigDecimal => Boolean, errorMessageKey: => String)(value: BigDecimal) =
    if (isValid(d)) Right(d) else Left(FormError(key, errorMessageKey))
```
But that is so short that it might be easier not to have this method in the first place

Reply:
I'm think you can combine this function and the one above it to suit how you're using it in both your scenarios:
```scala
def validateValueIsLessThan(key: String, limitInclusive: AmountInPence, errorMessageKey: => String)(value: BigDecimal) =
  if(value <= limitInclusive) Right(d) else Left(FormError(key, errorMessageKey))
```
Then you can use it for both your scenarios:
```scala
    .flatMap(value =>
        MoneyUtils.validateValueIsLessThan(
          lettingsReliefLimit.inPounds(),
          "lettingsRelief",
          ,
          "error.lettingsReliefLimit"
        )(value)
      )
      .flatMap { value =>
        MoneyUtils.validateValueIsLessThan(
          "lettingsRelief",
          privateResidencyRelief.inPounds()
        )(
          value
        )
```
 The functions basically do the same thing but slightly differently

Reply:
Oh, ok, done.